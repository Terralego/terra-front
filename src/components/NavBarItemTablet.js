import React from 'react';
import PropTypes from 'prop-types';

import HeaderLink from './HeaderLink';
import HeaderButton from './HeaderButton';

export const NavBarItemTablet = ({
  id,
  label,
  href,
  onClick,
  icon,
}) => (
  <HeaderLink
    href={href}
    onClick={onClick}
    data-link-id={id}
  >
    <p className="header-button-label">{label}</p>
    <HeaderButton
      id={id}
      icon={icon}
      alt={label}
    />
  </HeaderLink>
);

NavBarItemTablet.propTypes = {
  id: PropTypes.string.isRequired,
  href: PropTypes.string.isRequired,
  icon: PropTypes.string,
  label: PropTypes.string,
  onClick: PropTypes.func,
};

NavBarItemTablet.defaultProps = {
  icon: '',
  label: '',
  onClick () {},
};

export default NavBarItemTablet;
