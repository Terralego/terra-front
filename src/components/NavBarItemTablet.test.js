import React from 'react';
import renderer from 'react-test-renderer';
import { NavBarItemTablet } from './NavBarItemTablet';

jest.mock('./HeaderButton', () => () => 'HeaderButton');
jest.mock('./HeaderLink', () => ({ children }) => (
  <div title="headerlink">
    {children}
  </div>
));

it('should render correctly the component', () => {
  const tree = renderer.create(
    <NavBarItemTablet href="visualiser" id="test" />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
