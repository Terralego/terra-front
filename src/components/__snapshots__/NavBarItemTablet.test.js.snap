// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly the component 1`] = `
<div
  title="headerlink"
>
  <p
    className="header-button-label"
  >
    
  </p>
  HeaderButton
</div>
`;
